pkgs/development/compilers/gnu-smalltalk/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchurl, pkgconfig, libtool, zip, libffi, readline, gmp,
+gnutls, gnome, cairo, SDL, sqlite }:
+
+stdenv.mkDerivation rec {
+
+  version = "3.2.5";
+  name = "gnu-smalltalk-${version}";
+
+  src = fetchurl {
+    url = "mirror://gnu/smalltalk/smalltalk-${version}.tar.xz";
+    sha256 = "1k2ssrapfzhngc7bg1zrnd9n2vyxp9c9m70byvsma6wapbvib6l1";
+  };
+
+  # The dependencies and their justification are explained at
+  # http://smalltalk.gnu.org/download
+  buildInputs = [
+    pkgconfig libtool zip libffi readline gmp gnutls gnome.gtk cairo
+    SDL sqlite
+  ];
+
+  configureFlags = [ "--without-emacs" ];
+
+  checkPhase = "make check";
```

Comment:
Is this needed? Should be already doing `make check`. Add `doCheck = true;` instead.